src/charm.py

```diff
@@ -120,15 +142,29 @@ def _login_ui_layer(self) -> Layer:
         return Layer(pebble_layer)
 
     def _get_kratos_endpoint_info(self) -> Optional[str]:
-        kratos_public_url = ""
         if self.model.relations[self._kratos_relation_name]:
             try:
                 kratos_endpoints = self.kratos_endpoints.get_kratos_endpoints()
                 kratos_public_url = kratos_endpoints["public_endpoint"]
+                return kratos_public_url
             except KratosEndpointsRelationDataMissingError:
                 logger.info("No kratos-endpoint-info relation data found")
 
-        return kratos_public_url
+    def _update_login_ui_endpoint_relation_data(self, event: RelationEvent) -> None:
+        endpoint = (self.ingress.url if self.ingress.is_ready() else "",)
+
+        logger.info(f"Sending login ui endpoint info: endpoint - {endpoint[0]}")
+
+        self.endpoint_provider.send_endpoints_relation_data(endpoint[0])
+
+    def _get_hydra_endpoint_info(self) -> Optional[str]:
+        if self.model.relations[self._hydra_relation_name]:
```

Comment:
we need an `else` statement in case there is no relation with hydra

Reply:
Fair, but I'm not sure what default value it should return. For now I'll send back None and open an issue in the login-ui server to patch in error handling in case the hydra and kratos endpoints are inaccessible.

Reply:
I would return an empty string instead

Reply:
done, returning empty str instead